service/controller/resource/tcnp/create.go

```diff
@@ -212,11 +245,17 @@ func (r *Resource) updateStack(ctx context.Context, cr infrastructurev1alpha2.AW
 	{
 		r.logger.LogCtx(ctx, "level", "debug", "message", "requesting the update of the tenant cluster's node pool cloud formation stack")
 
+		tags, err := r.getCloudFormationTags(ctx, cr)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
 		i := &cloudformation.UpdateStackInput{
 			Capabilities: []*string{
 				aws.String(capabilityNamesIAM),
 			},
 			StackName:    aws.String(key.StackNameTCNP(&cr)),
+			Tags:         tags,
```

Comment:
No, all resources that support tags update this value. but in case of the ASG the machines are only updated when new instances are added to the group

```
 If you modify a tag, the updated version of the tag is added to instances launched in the Auto Scaling group after the change. If you create or modify a tag for an Auto Scaling group, these changes are not made to instances that are already running in the Auto Scaling group.
```
https://docs.aws.amazon.com/autoscaling/ec2/userguide/autoscaling-tagging.html#add-tags

Anyway, I tested multiples times. Also, I check there is no interruption having a pingdom checking service availability of helloworld. So all changed on resources like SG or Subnet does not impact workload traffic

Reply:
I think there could be corner cases that this would trigger a node roll, if there is already some changes on CR  that are not applied yet but would cause node roll than this harmless Tag update would trigger the changes as well.

Reply:
But if there is a change in node pool CR, it should not be the case? I mean, when you change the CR spec, operator should trigger the changes, right?